### ISSUE TITLE
Textidote - spelling check

### ISSUE THREAD

**Issue by Princeofthebow** [lervag/vimtex#1857]
Hi sorry it is me again from yesterday!
As I'm navigating my way trough I'm now looking to make use of a spell checker. From the documentation from what I understand using textidote is possible solution.

I have therefore set the followin `.vimrc` file:

```
call plug#begin('~/.vim/plugged')
Plug 'lervag/vimtex'
call plug#end()

let g:tex_flavor = 'latex'
let g:vimtex_view_method = 'mupdf'
let g:vimtex_grammar_textidote = {
            \ 'jar': '/usr/local/bin/textidote.jar',
            \ 'args': '--check en',
            \}
```

This is my MWE tex file:

```
\documentclass{article}
\begin{document}
Hellor Worlsd!
\end{document}

```

but after running `:compiler textidote | lmake`
I get the following outcome:

```
make: *** No targets specified and no makefile found.  Stop.

Press ENTER or type command to continue
make: *** No targets specified and no makefile found.  Stop.

```
and then after i hit enter an empty vim with:

```
(1 of 1): make: *** No targets specified and no makefile found.  Stop.
```
This has puzzled me since I as mentioned I'm rather new to vim and vimtex (but not latex). 
I have run textidote on the textfile and it appears to be working correctly

ps is there any other way to handle spellcheck that may be better?

Thanks in advance!

**Comment by lervag** [comment]
Can you show the output of `:echo &makeprg` and `echo &errorformat` after you do `:compiler textidote` (without `lmake`!)?

**Comment by Princeofthebow** [comment]
I get the following error message:
`
vimtex: g:vimtex_grammar_textidote is not properly configured!
        Please see ":help vimtex-grammar-textidote" for more details.
`
Hum - I wonder why I get this error as it appears that the config in my vimrc file is correct. :-(

`:echo &makeprg`only prints `make`

while `:echo &errorformat` gives me

```
%*[^"]"%f"%*\D%l: %m,"%f"%*\D%l: %m,%-G%f:%l: (Each undeclared identifier is reported only once,%-G%f:%l: for each function it appears in.),
%-GIn file included from %f:%l:%c:,%-GIn file included from %f:%l:%c\,,%-GIn file included from %f:%l:%c,%-GIn file included from %f:%l,%-G%
*[ ]from %f:%l:%c,%-G%*[ ]from %f:%l:,%-G%*[ ]from %f:%l\,,%-G%*[ ]from %f:%l,%f:%l:%c:%m,%f(%l):%m,%f:%l:%m,"%f"\, line %l%*\D%c%*[^ ] %m,%
D%*\a[%*\d]: Entering directory %*[`']%f',%X%*\a[%*\d]: Leaving directory %*[`']%f',%D%*\a: Entering directory %*[`']%f',%X%*\a: Leaving dir
ectory %*[`']%f',%DMaking %*\a in %f,%f|%l| %m
```

**Comment by lervag** [comment]
I think the problem is that your configuration is not correct.

```vim
let g:vimtex_grammar_textidote = {
            \ 'jar': '<PATHTO>\textidote.jar',
            \ 'args': '--check en',
            \}
```

The path in the `.jar` key is not valid. I would write the path like this: `/path/to/textidote.jar`. Not sure if the backslash is itself the reason.

**Comment by Princeofthebow** [comment]
This is what I have in the config, i.e. the whole path:

```
let g:vimtex_grammar_textidote = {
            \ 'jar': '/usr/local/bin/textidote.jar',
            \ 'args': '--check en',
            \}
```
I had omitted it in the previous post. BTW textidote can be called even from $HOME

**Comment by Princeofthebow** [comment]
Ok I confim that the problem was indeed the installation of Textidote as via the jar file it works with `:compiler textidote|lmake`

Perhaps highlighting this in the documentation might be a good idea as I was convinced the the *.deb file would have placed me the textidote jar file somewhere on system but it did not and in particular it was not a jar file.

I'm starting to have a nice workflow now - once again excellent job on both sw and support. consider that I'm a novice and now I cannot event _think_ of going back to winedt.

**Comment by lervag** [comment]
I'm not quite sure I understand what you mean: did you first try to specify a textidote executable (e.g. `/usr/bin/textidote`), i.e. not the jar file specifically?

I'm not quite sure exactly what has been confusing, but I don't mind updating and improving the docs. Do you have a concrete suggestion?

> I'm starting to have a nice workflow now - once again excellent job on both sw and support. consider that I'm a novice and now I cannot event _think_ of going back to winedt.

Thanks, I'm glad to hear it! :) Btw, I've never heard of WinEdt before (http://www.winedt.com/). There's several other dedicated LaTeX editors, e.g. texmaker and texsuite; I've heard a lot of people find these to be good. But I guess once you go Vim (or neovim), you don't go back :)

**Comment by Princeofthebow** [comment]
Perhaps I have been unclear,sorry: Texidote for Debian (the distro I use) offers a *.deb package (see [here](https://github.com/sylvainhalle/textidote/releases)). Such package does not place the texitidote jar file anywhere(i.e./usr/bin/textidote was not a jar file that vimtex could fetch).
I then downloaded the jar file (not the *.deb package) and after pointing vimrc to the right folder (where I had placed the textidote jar file) it all started working.

In other words the *.deb package of textidote did not work and maybe highlighting this on the docs could be a good idea. I.e. specify that the actual jar file is needed

Vim for the win!

**Comment by lervag** [comment]
Thanks, I've updated the docs. Do you think it suffices?

**Comment by ChasingZenith** [comment]
> Texidote for Debian (the distro I use) offers a *.deb package (see [here](https://github.com/sylvainhalle/textidote/releases)). Such package does not place the texitidote jar file anywhere(i.e. /usr/bin/textidote was not a jar file that vimtex could fetch - I used `whereis` to search the file).

Hi @Princeofthebow, I am a Manjaro user and I find that `textidote.jar` exists on my computer even if I install `textidote` with a repository. Since I think that package in AUR repository seems to be based on the .deb package. I think these two Linux distributions may have similar behavior.

In fact when I look at what `/usr/bin/textidote` is, I find that it is a `bash` script that wraps up `textidote.jar`.

```bash
#! /bin/bash
java -jar /opt/textidote/textidote.jar --name textidote "$@"
```

So I think that the following configuration might also works for you 
```vim
let g:vimtex_grammar_textidote = {
            \ 'jar': '/opt/textidote/textidote.jar',
            \ 'args': '',
            \}
```

BTW, in this process, I find that the first time I run `:compiler textidote|lmake`, nvim gives me an error which says that function `s:get_textidote_lang` is unknown. But execute it again textidote can do proof-reading (but no quickfix windows pop up but just like something were excuted in the integral terminal).

```
Error detected while processing /home/user/.cache/vimfiles/repos/github.com/lervag/vimtex/compiler/textidote.vim:
line   24:
E117: Unknown function: s:get_textidote_lang
E15: Invalid expression: 'java -jar ' . shellescape(fnamemodify(s:cfg.jar, ':p')) . (has_key(s:cfg, 'args') ? ' ' . s:cfg.args : '') . ' --no-color --
textidote_lang(s:language) . ' %:S'
/opt/textidote/textidote.jar
```

 I look at `vimtex/compiler/textidote.vim` and find that this function is called before it is defined. Is it the reason why it happened? ( perhaps I need to open a new Issue? @lervag

**Comment by lervag** [comment]
> So I think that the following configuration might also works for you
> 
> ```viml
> let g:vimtex_grammar_textidote = {
>             \ 'jar': '/opt/textidote/textidote.jar',
>             \ 'args': '',
>             \}
> ```

I can confirm that this should work well on Arch Linux.

> BTW, in this process, I find that the first time I run `:compiler textidote|lmake`, nvim gives me an error which says that function `s:get_textidote_lang` is unknown.

Thanks, fixed now!

**Comment by ChasingZenith** [comment]
Thank you for your instant response!!!

BTW, is that no quickfix window the normal way it behave?

**Comment by lervag** [comment]
No problem! :)

And yes, it is. See `:help vimtex-lint` (as suggested in intro of `:help vimtex-grammar`) for an introduction on how to use this feature. The idea is to rely on the built-in feature as much as possible.

**Comment by ChasingZenith** [comment]
Thank you. I will read it later.
